docs/reference/argocd.md

```diff
@@ -155,6 +155,10 @@ AppSync | 3m | AppSync is used to control the sync frequency of ArgoCD Applicati
 Sharding.enabled | false | Whether to enable sharding on the ArgoCD Application Controller component. Useful when managing a large number of clusters to relieve memory pressure on the controller component.
 Sharding.replicas | 1 | The number of replicas that will be used to support sharding of the ArgoCD Application Controller.
 Env | [Empty] | Environment to set for the application controller workloads
+Sharding.dynamicScalingEnabled | true | Whether to enable dynamic scaling of the ArgoCD Application Controller component. This will ignore the configuration of `Sharding.enabled` and `Sharding.replicas`
+Sharding.minShards | 1 | The minimum number of replicas of the ArgoCD Application Controller component.
```

Comment:
Can we mention the validation critierias in a separate column ?
- ClustersPerShard, MinShards and MaxShards must be 1 or > 1
- MaxShards must always be greater than MinShards